### ISSUE TITLE
Remove ArrayAccess from the Model

### ISSUE THREAD

**Issue by stanistan** [SkyPHP/skyphp#157]
`Model` would no longer implement `ArrayAccess` and get rid of `ModelArrayObject`, so that plural subobjects in a model are accessed as arrays.

This would change:

``` php

AQL:
artist {
    name,
    [artist_album]s,
    artist_ablum as album_sub {
        artist_id,
        name
    }
}

<?php

$o = new artist;

// from
$o->artist_album[0]->name == $o['artist_album'][0]['name']; // true
$o->album_sub[0]->name == $o['artist_sub'][0]['name']; // true

//to
$o->artist_album[0]->name; // can no longer be accessed as an array
$o->artist_sub[0]['name'];  // can no longer be accessed as an object
```

This makes the way one accesses a `Model` object property imply what kind of data it is.

**Comment by tolijoonbug** [comment]
I agree. And then we don't have to do count() to check if it's empty

**Comment by will123195** [comment]
resolved in 3.0
